docs/meshstack.customer-group-sync.md

```diff
@@ -3,142 +3,192 @@ id: meshstack.customer-group-sync
+          A filter that follows the LDAP search filter format.
+    -}
+      { attributes : Text, base : Text, filter : Text }
 
-<!--snippet:mesh.identityconnector.ldap.filter#type-->
+let CollectorConfiguration =
+    {-
+      pageSize:
+          The size of a single page that will be returned as a result of an LDAP search
 
+      group:
+          LdapFilter which will contain the information needed to search the LDAP source for groups.
 
-<!--DOCUSAURUS_CODE_TABS-->
-<!--Dhall Type-->
-```dhall
-let LdapFilter =
-    {-
-        attributes:
-            A comma separated list of attributes that should be returned per LDAP entry.
-            For example, "cn, uniqueMember, description". These attributes can later be
-            used to populate the value of a meshObject field.
-        base: The base from which the query should be performed. For example, "ou=groups"
-        filter: A filter that follows the LDAP search filter format. For example, "'(cn=mc*)'"
+      user:
+          LdapFilter which will contain the information needed to search the LDAP source for users.
     -}
-      { attributes : Text, base : Text, filter : Text }
+
+      { pageSize : Natural, group : LdapFilter, user : LdapFilter }
+```
+<!--Example-->
+```dhall
+let example
+    : CollectorConfiguration
+    = { pageSize = 100
+      , group =
+        { attributes = "cn, uniqueMember, description"
+        , base = "ou=groups"
+        , filter = "'(cn=*)'"
+        }
+      , user =
+        { attributes = "entryDN, uid, sn, givenName, mail"
+        , base = "ou=people"
+        , filter = "'(uid=*)'"
+        }
+      }
 ```
 <!--END_DOCUSAURUS_CODE_TABS-->
 
 
-The following configuration describes how the collected data should be transformed into meshObjects.
-<!--snippet:mesh.identityconnector.ldap.transformation#type-->
+### Transform
+
+In the Transform step, the collected data is transformed into meshObjects.
+
+An `AttributeTransformation` will specify an LDAP attribute and how to transform it before assigning to a meshObject field.
+
+<!--snippet:mesh.identityconnector.ldap.attributeTransformation#type-->
 
 
 <!--DOCUSAURUS_CODE_TABS-->
 <!--Dhall Type-->
 ```dhall
-let TransformationConfiguration =
+let AttributeTransformation =
     {-
-    mesh-object:
-      The type of meshObject this configuration is about. Can be one of
-      "MeshUser", "MeshCustomerUserGroup" or "MeshCustomerGroupBinding"
-    fields:
-      A list of configuration objects specifying which LDAP attribute gets mapped to which
-      meshObject attribute, and whether any transformations should be done on the attribute before
-      assigning it to the meshObject field.
-    tags: Same as fields above, but for tags on the specified meshObject.
+      An AttributeTransformation is used to define an attribute of an LDAP entry and how to transform it so that it
+      can be assigned to a field in a meshObject. There are two types of AttributeTransformations, Static and Regex.
```

Comment:
can you provide an example like group name?

Reply:
A complete example are given in the dhall example sections below.

Reply:
@pubudug I'm for keeping examples in the code tabs for individual types (there's a Type + Example tab)

It's great if you give a full end to end example for identity connector configuration, however to me that's more of a "how to" than a reference documentation

Reply:
Yeah. This PR is organized so that the examples show in it's own tab.